packages/graph/hash_graph/lib/graph/src/api/rest/data_type.rs

```diff
@@ -93,8 +94,15 @@ async fn create_data_type<P: StorePool + Send>(
         StatusCode::INTERNAL_SERVER_ERROR
     })?;
 
+    let data_type: DataType = schema.try_into().into_report().map_err(|report| {
+        tracing::error!(error=?report, "Couldn't convert schema to Property Type");
+        StatusCode::UNPROCESSABLE_ENTITY // TODO - Do we want to return a 400 instead
```

Comment:
Not sure about this return code, I was [reading this](https://stackoverflow.com/questions/49648442/http-status-codes-when-errors-in-the-request-body). Still unsure

Reply:
I believe Axum uses when its extractors fail to deserialize, so this seems like a valid error code to use

Reply:
I guess my uncertainty here is that this happens during _parsing_ of the object rather than deserialization. So technically the JSON they gave us was correctly formed according to the structural constraints of the schema, it was just additional validation logic which failed.

Reply:
I've removed the TODO